ocs_ci/ocs/ocp.py

```diff
@@ -106,6 +106,33 @@ def exec_oc_cmd(self, command, out_yaml_format=True, secrets=None, **kwargs):
             return yaml.safe_load(out)
         return out
 
+    def exec_oc_debug_cmd(self, node, cmd_list):
```

Comment:
why not enhance exec_oc_cmd(), :)

Reply:
generally for minor variances, its better to enhance existing functions,  I also brought this up with mcg stuff.

Reply:
@vasukulkarni, `debug` command execution will create a pod and executes the command inside the pod. 
In debug cmd execution failure is captured using `f" || echo 'CMD FAILED';".join(cmd_list)` and it supports to run multiple oc commands inside the debug pod. 
To enhance existing oc_exec_cmd, might need to add multiple if conditions before(for checking debug mode) and after(to check the error output).

Reply:
@vasukulkarni if we will keep up enhancing general functions like `run_cmd` or `oc_exec_cmd` by other use cases it will make the general functions much complex, we will have to introduce many conditions inside, it will needs to have much more parameters or parameters which will needs to take multiple types which is not good practice.

The function should be simple and straightforward which in our case the `run_cmd` or `exec_oc_cmd` they are. What advantage we will get when we will make them more complex for other use cases and will add additional parameters? I don't see any IMHO.

If we have some other use cases like this I prefer separate function for specific use case than adding complexity to the current function which is broadly used which can easily lead to the failures of current code. 

If this change will introduce some bug because of complexity because we added more parameters and conditions inside it will be hard to spot all the possible variants of parameters also in case we will add more and more parameters as you suggest by enhancing general function in the future it will make it even more complex which is not worth IMO.

Reply:
@petr-balogh The code changes for this are so minimial, infact just 2 lines changed w.r.t exec_oc_cmd

```python
        cmd = f" || echo '{err_msg}';".join(cmd_list)
        debug_cmd = f"debug nodes/{node} -- chroot /host /bin/bash -c \"{cmd}\""
```

and probably with some simple changes to exec_oc_cmd and handling of list cmds, eg:


```python
        if 'debug' in command:
            cmd = f" || echo '{err_msg}';".join(cmd_list)
            oc_cmd = f"debug nodes/{node} -- chroot /host /bin/bash -c \"{cmd}\""
```
we can take advantage of smaller set of code and benefit from error handling in one place

If debug is doing drastically different from the other exec_cmd it might make sense to move as different method, but with above 3 or 4 line changes, I dont see any complexity or readability issues you are mentioning.

Having said that, I am not holding this PR for now, but going to bring this up and make sure we dont right too many `exec_specific_cmd` routines unless strictly needed.